src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java

```diff
@@ -694,9 +694,27 @@ public void run() {
 		final int lastVoteDirection = voteDirection;
 
 		switch(action) {
-			case DOWNVOTE: voteDirection = -1; break;
-			case UNVOTE: voteDirection = 0; break;
-			case UPVOTE: voteDirection = 1; break;
+			case DOWNVOTE:
+				if(!src.archived) {
+					voteDirection = -1;
+				} else {
+					voteDirection = lastVoteDirection;
```

Comment:
I think these else clauses can be removed -- {{voteDirection}} doesn't need to be set at all here.